ssl-config/src/main/java/com/palantir/remoting/ssl/KeyStores.java

```diff
@@ -26,40 +29,97 @@
 import java.nio.file.Path;
 import java.security.GeneralSecurityException;
 import java.security.Key;
+import java.security.KeyFactory;
 import java.security.KeyStore;
+import java.security.PrivateKey;
 import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
 import java.security.cert.CertificateFactory;
+import java.security.spec.RSAPrivateKeySpec;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 final class KeyStores {
 
+    /**
+     * Pattern that matches a single RSA key in a PEM file. Has a capture group that captures
+     * the content of the key (everything that occurs between the header and footer).
+     */
+    private static final Pattern KEY_PATTERN = Pattern.compile(
+            "-----BEGIN RSA PRIVATE KEY-----(.+)?-----END RSA PRIVATE KEY-----", Pattern.DOTALL);
```

Comment:
Could make it optional in the regex :)

On Mon, Apr 11, 2016 at 12:06 PM Nick Miyake <notifications@github.com>
wrote:

> In ssl-config/src/main/java/com/palantir/remoting/ssl/KeyStores.java
> <https://github.com/palantir/http-remoting/pull/92#discussion_r59262397>:
>
> >
> >  final class KeyStores {
> >
> > +    /**
> > +     * Pattern that matches a single RSA key in a PEM file. Has a capture group that captures
> > +     * the content of the key (everything that occurs between the header and footer).
> > +     */
> > +    private static final Pattern KEY_PATTERN = Pattern.compile(
> > +            "-----BEGIN RSA PRIVATE KEY-----(.+)?-----END RSA PRIVATE KEY-----", Pattern.DOTALL);
>
> Ah, got it. Yeah right now the code strips all newlines so didn't have
> that, but looks like newline after header and before footer is mandated by
> the format, so should be able to bake into the regex. Will do.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/palantir/http-remoting/pull/92/files/778de38d91deecec8d1d8164d58699adb3ffdd41#r59262397>
>